docs/content/guides/developer/app-examples/reviews-rating.mdx

```diff
@@ -0,0 +1,212 @@
+
+### Moderators
+
+Moderators monitor content of the reviews and can delete any reviews that contain inappropriate content.
+Moderators also keep an eye on the number of reviews registered for a service to ensure that they do not exceed the maximum number allowed.
+When the maximum number of reviews is reached, moderators can prune old reviews to make room for a new review.
+
+## How reviews are rated
+
+The reviews are rated (scored) on-chain using following criteria
+- Intrinsic score (IS)
+  - Length of content
+- Extrinsic score (ES)
+  - Number of votes received
+- Verification multiplier (VM)
+  - Reviews with PoE get a multiplier to improve rating
+
+```
+Total Score = (IS + ES) * VM
+```
+
+## Smart contracts
+
+There are several modules that create the backend logic for the example. 
+
+### dashboard.move
+
+The `dashboard.move` module defines the `Dashboard` struct that is used to group services.
+
+```move
+struct Dashboard has key, store {
+  id: UID,
+  set: VecSet<ID>,
+  service_type: String
+}
+```
+
+The services are grouped by attributes, which can be cuisine type, geographical location, operating hours, google maps ID, and so on. To keep it basic, the example stores only `service_type` (for example, fast food, Chinese, Italian).
+
+### service.move
+
+This module defines the `Services` struct, which service owners manage.
+
+```move
+struct Service has key, store {
+  id: UID,
+  reward_pool: Balance<SUI>,
+  reward: u64,
+  reviews: MultiMap<ID>,
+  moderators: VecSet<address>,
+  overall_rate: u64,
+  name: String
+}
+```
+
+All the submitted reviews are stored in `reviews` field, which has `MultiMap<ID>` type. The [multimap module](#multimap) defines the type. 
+
+The `reward_pool` and `reward` fields store the available reward information.  
+
+```move
+struct AdminCap has key, store {
+  id: UID,
+  service_id: ID
+}
+struct Moderator has key {
+  id: UID,
+  service_id: ID,
+}
+```
+`SERVICE OWNERS` are given `AdminCap` and `MODERATORS` are given `Moderator` to allow them to perform previleged operations.
+
+### multimap.move {#multimap}
+
+`MultiMap` is an implementation of the `multimap` data structure (https://en.wikipedia.org/wiki/Multimap), which keeps all the reviews sorted. Using `MultiMap`, all the reviews are ordered by rating, with the highest rated reviews coming first.
+
+```move
+struct MultiMap<K: copy> has copy, drop, store {
+  contents: vector<Entry<K>>,
+}
+struct Entry<K: copy> has copy, drop, store {
+  key: K,
+  priority: u64,
+}
+```
+The `object_id` of a `review` object is used as a `key` and `total_score` is used as `priority` for a new entry.
+
+Note that `vector` is used to store the contents of the `MultiMap`. This is acceptable in our use case because the maximum number of reviews for a service is 1000.
+As reviews are sorted by `total_score` on-chain, it is necessary to keep a manageable number of reviews for a service.
```

Comment:
I don't believe you need to store all reviews in sorted order on-chain, you just need to store the top `K` reviews that are going to be rewarded, and have a way to move reviews in and out of this list based on the votes they get.

Reply:
I modified the code so that it only maintains top rated reviews for distributing rewards.